src/ponyc/main.c

```diff
@@ -247,6 +247,34 @@ static bool compile_package(const char* path, pass_opt_t* opt,
   return ok;
 }
 
+static char* compiler_version()
+{
+#ifdef PLATFORM_IS_WINDOWS
+  // TODO get C compiler info on Windows
+  return "";
+#else
+  char* cc = PONY_COMPILER;
+  char* args = " --version";
+  char* cmd = malloc(strlen(cc) + strlen(args));
```

Comment:
The allocation should use the Pony pool allocator for consistency with the rest of the memory allocation in the compiler.